### PR TITLE
Making database encoding functions available in pg_proc

### PR DESCRIPTION
**Pull request by kuien**
Three functions are available now:
- convert_from(bytea, name) returns text
- convert_to(text, name) returns bytea
- length(bytea, name) returns int

Related commits (backported before OSS but not actived)

    commit a9742f123c25c7bc5ce0c18e188dba790da4fbe1
    Author: Tom Lane <tgl@sss.pgh.pa.us>
    Date:   Wed Jan 9 23:43:54 2008 +0000

        Remove incorrect (and ill-advised anyway) pfree's in pg_convert_from and
        pg_convert_to.  Per bug #3866 from Andrew Gilligan.

    commit 55613bf9cd7d6071e43e68ac14bc0243a1027507
    Author: Andrew Dunstan <andrew@dunslane.net>
    Date:   Tue Sep 18 17:41:17 2007 +0000

        Close previously open holes for invalidly encoded data to enter the
        database via builtin functions, as recently discussed on -hackers.

        chr() now returns a character in the database encoding. For UTF8 encoded databases
        the argument is treated as a Unicode code point. For other multi-byte encodings
        the argument must designate a strict ascii character, or an error is raised,
        as is also the case if the argument is 0.

        ascii() is adjusted so that it remains the inverse of chr().

        The two argument form of convert() is gone, and the three argument form now
        takes a bytea first argument and returns a bytea. To cover this loss three new
        functions are introduced:
        . convert_from(bytea, name) returns text - converts the first argument from the
          named encoding to the database encoding
        . convert_to(text, name) returns bytea - converts the first argument from the
          database encoding to the named encoding
        . length(bytea, name) returns int - gives the length of the first argument in
          characters in the named encoding

### CODE REVIEW

**Comment by atris** [comment]
LGTM

File: src/backend/utils/mb/mbutils.c [hunk 1]
```diff
@@ -441,64 +441,6 @@ length_in_encoding(PG_FUNCTION_ARGS)
 }
 
 /*
- * Convert string using encoding_name.
- *
- * TEXT convert2(TEXT string, NAME src_encoding_name, NAME dest_encoding_name)
- */
-Datum
```
Comment (xiong-gang) [inline src/backend/utils/mb/mbutils.c]: why remove this function?

**Comment by xiong-gang** [comment]
since the old version of pg_convert is problematic, can we add some test cases?

**Comment by andreasscherbaum** [comment]
Except the test cases: +1

**Comment by kuien** [comment]
We retrieve related JIRAs on function convert*():
https://jira.eng.pivotal.io/browse/MPP-14759
https://jira.eng.pivotal.io/browse/MPP-17314
https://jira.eng.pivotal.io/browse/MPP-17340
https://jira.eng.pivotal.io/browse/MPP-17341
We read through these JIRAs and confirmed those issues are fixed by this commit.

@xiong-gang @andreasscherbaum Yes, we were considering to add more test cases, but there are few on upstream. Now we find a set of test cases from Ashwin Kumar's commits (https://fisheye.eng.pivotal.io/changelog/GPDB-ACE?cs=aaaaf4c0e2aa9c59c12d05eed542e9c70f9dc57e), we will add them asap.

**Comment by kuien** [comment]
@xiong-gang @andreasscherbaum tests are added, please have a look, thanks.

**Comment by yydzero** [comment]
LGTM

**Pull request merged**